src/Features/LanguageServer/Protocol/Handler/Completion/CompletionResolveHandler.cs

```diff
@@ -74,53 +74,79 @@ public async Task<LSP.CompletionItem> HandleRequestAsync(LSP.CompletionItem comp
                 }
             }
 
-            var selectedItem = list.Items.FirstOrDefault(i => i.DisplayText == data.DisplayText);
+            var selectedItem = list.Items.FirstOrDefault(i => i.DisplayTextPrefix + i.DisplayText + i.DisplayTextSuffix == data.CompleteDisplayText);
```

Comment:
Unfortunately the string concats will cause allocations on a per-completion item basis. Maybe that's ok but thought I'd point out!

Reply:
> Ah, good point. I switched the implementation to StringBuilder instead, hopefully that's more efficient?

Actually that might be less efficient because you'll then be allocating a string builder in addition to allocating the string. The most performant route would probably be to utilize STartSWith/EndsWith for the prefix/suffix and string.Compare for the mid-section.

Reply:
Ah you utilize the same stringbuilder. In that case it'd only be slightly more efficient because you'd still be allocating the string.